### ISSUE TITLE
Training time

### ISSUE THREAD

**Issue by Piotr94** [csbhr/CDVD-TSP#23]
Hi! Could you share information about training time for DVD dataset? On my GPU one epoch takes more than one hour and I wonder if it is correct.

**Comment by Etienne66** [comment]
Yes. I'd be interested in that as well. I'm doing a rather large data set (333 videos) and it is taking over 11 hours for a complete epoch (train + test).

Also in my case a batch_size of 2 takes up around 6GB of GPU memory.

**Comment by yzleroy** [comment]
> Hi! Could you share information about training time for DVD dataset? On my GPU one epoch takes more than one hour and I wonder if it is correct.

I use 4 x P40 (batch size = 32), and one epoch takes more than 1 hour. But whole training epochs are 500. Whole training time will take several weeks. I think it's difficult to finish whole training process.

**Comment by csbhr** [comment]
For our training settings, we set the batch size to 8 and used one RTX TITAN GPU. It took about 45 mins per epoch.

The results released in the Paper were trained for 300 epochs.

**Comment by Etienne66** [comment]
> 
> 
> For our training settings, we set the batch size to 8 and used one RTX TITAN GPU. It took about 45 mins per epoch.
> 
> The results released in the Paper were trained for 300 epochs.

This card has 24GB of RAM. I plan on buying an RTX 3090 (also 24GB) next year myself. Glad to know that it should do a bit better than the card @csbhr used.

On an RTX 2060 Super with 8GB of RAM I can only do a batch size of 3 and it takes about 2 hours per epoch on the DVD data set.

**Comment by yuanze-lin** [comment]
> > For our training settings, we set the batch size to 8 and used one RTX TITAN GPU. It took about 45 mins per epoch.
> > The results released in the Paper were trained for 300 epochs.
> 
> This card has 24GB of RAM. I plan on buying an RTX 3090 (also 24GB) next year myself. Glad to know that it should do a bit better than the card @csbhr used.
> 
> On an RTX 2060 Super with 8GB of RAM I can only do a batch size of 3 and it takes about 2 hours per epoch on the DVD data set.

Will you still try to reproduce whole training process, since the training time is too long ?

**Comment by Etienne66** [comment]
> 
> 
> > > For our training settings, we set the batch size to 8 and used one RTX TITAN GPU. It took about 45 mins per epoch.
> > > The results released in the Paper were trained for 300 epochs.
> > 
> > 
> > This card has 24GB of RAM. I plan on buying an RTX 3090 (also 24GB) next year myself. Glad to know that it should do a bit better than the card @csbhr used.
> > On an RTX 2060 Super with 8GB of RAM I can only do a batch size of 3 and it takes about 2 hours per epoch on the DVD data set.
> 
> Will you still try to reproduce whole training process, since the training time is too long ?

Nope. I'm trying to do better and faster. I have rewritten a good portion of the code to run faster and I've changed the code to use Super-Convergence. I'm hoping to get it down to a week or two with better results in about 100 to 150 Epochs on the DVD data. After I'm sure that works I'll combine the DVD, GOPRO, and REDS data sets to see if that improves it any more. I figure with all of that data it will take no more than 1 month to finish a Super-Convergence training and get better results.

Take a look at the Wiki for my fork of this project. I finished Epoch 100 with a PSNR of 31.149 and SSIM of 0.912. Not as good as I would have liked. I should have set it to 150 Epochs.
![psnr](https://user-images.githubusercontent.com/52262160/125176368-7cccb080-e198-11eb-9c7f-8d6de448d10d.png)
![ssim](https://user-images.githubusercontent.com/52262160/125176370-7d654700-e198-11eb-8f7f-7af608bc76a1.png)

**Comment by Etienne66** [comment]
I had trained it with 3 frames at a patch size of 720. While the training went good and I did get stats close to the original paper when I tested this model trained at Stage 1 it did not work well at Stage 2.

It appears that the models work best at the Stage they were trained for. However I did try the author's Stage 2 trained model and tried it at Stage 1. The results were almost identical to his Stage 1 trained model.

I wish I had a card where I could train a model at stage 3 with a patch size of 720. The most I can get is 320 and it will take a long time to train that with my hardware. See [my MM-SSIM_Loss branch](https://github.com/Etienne66/CDVD-TSP/tree/MM-SSIM_Loss) and [Wiki](https://github.com/Etienne66/CDVD-TSP/wiki) for more details including logs.
